### ISSUE TITLE
HTTP 400 kills the replicator and keeps all the user data hostage

### ISSUE THREAD

**Issue by sellingsolutions** [couchbase/couchbase-lite-ios#1279]
I've collected some logs for you guys to use, they're in a dropbox folder that I'll share with you separately. 

```
20:54:38.315‖ WARNING: CBLRestPusher[https://userid:*****@domainname/dbname]: _bulk_docs got an error: {
    error = "bad_request";
    id = "document_1450347529_c6336ee8-eede-40cd-be74-d88f184f0559";
    reason = "Missing digest in stub attachment \"utl\U00e5tande.doc\"";
    status = 400;
}
20:54:38.350‖ Sync: CBLRestPusher[https://userid:*****@domainname/dbname] Progress: set error = 400 bad_request
20:54:38.350‖ Sync: CBLRestPusher[https://userid:*****@domainname/dbname] STOPPING...
20:54:38.350‖ Sync: Stopping 128 remote requests
```
--------
* Version:   CouchbaseLite 1.2.1
* Client OS: iPad Air 2 - iOS 9.3
* Couchbase Sync Gateway/1.2.1(4;26c202a)
* Couchbase Server 4.1.1-5914 Enterprise Edition (build-5914)

**Comment by snej** [comment]
"Missing digest in stub attachment " sounded very very familiar — [searching the SG issues](https://github.com/couchbase/sync_gateway/issues?utf8=✓&q=Missing+digest+in+stub+attachment) turns up https://github.com/couchbase/sync_gateway/issues/1706 which was fixed in April for 1.3.

**Comment by sellingsolutions** [comment]
Cool, I'll try 1.3 on the database that I shared with you via dropbox!
